spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigMapPropertySource.java

```diff
@@ -104,7 +90,7 @@ else if (propertyName.endsWith(".properties")) {
 
 		return input.entrySet().stream().map(e -> extractProperties(e.getKey(), e.getValue(), environment))
 				.filter(m -> !m.isEmpty()).flatMap(m -> m.entrySet().stream())
-				.collect(Collectors.toMap(Entry::getKey, Entry::getValue, throwingMerger(), LinkedHashMap::new));
+				.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, throwingMerger(), HashMap::new));
```

Comment:
although, I did some changes in this PR and some others where I removed `throwingMerger(), LinkedHashMap::new`, I think that these are justified here. I prefer a failure, rather then an override, from diff properties of `ConfigMap`.